docs/blueprint-designer-guide/blueprints.md

```diff
@@ -379,3 +392,107 @@ grains:
+
+### source 
+Please see [the grain source](blueprints.md#source) for more details.
+
+### host
+Please see [the grain host](blueprints.md#host) for more details.
+
+### inputs​
+Similar to blueprint inputs, CloudFormation grain inputs allow you to reuse the same CloudFormation module in different ways. Inputs provided to the CloudFormation grain are used when launching the CloudFormation module. CloudFormation grain inputs should be listed in the order defined in the module's _variables.tf_ file.
+
+### tags​
+Whenever a CloudFormation grain is launched, all resources created during the deployment process are automatically tagged with Torque's system tags, blueprint tags and customer tags.
+
+### outputs​
+Outputs are strings generated by CloudFormation during the deployment process.
+
+```yaml"
+grains:
+  database:
+    kind: cloudformation
+    spec:
+      source:
+        path: github.com/org/repo.git//terraform/rds
+        ...
+      outputs:
+        - hostname
+        - connection_string
+```
+
+## The Kubernetes Grain​
+The Kubernetes grain allows you to use native Kubernetes manifests, manifest catalogs in a given user's repository. Currently, the solution has restriction: it is not possible to launch concurrent sandboxes with Kubernetes grain (because the manifest resources are static and their names are not unique).
+The deployment namespace must exist in the cluster prior to the deployment. It must not be equal to the namespaces used by Torque for agent deployments.
+
+### source 
+Please see [the grain source](blueprints.md#source) for more details.
+
+### host
+Please see [the grain host](blueprints.md#host) for more details.
+
+### tags​
+Whenever a Kubernetes grain is launched, all resources created during the deployment process are automatically tagged with Torque's system tags, blueprint tags and customer tags. if you wish to disable tagging for all resources in a specific Kubernetes grain, use the following syntax:
+
+```yaml"
+grains:
+  database:
+    kind: kubernetes
+    spec:
+      tags:
+        auto-tag: false
+```
+
+### namespace
+The deployment namespace must exist in the cluster prior to the deployment. It must not be equal to the namespaces used by Torque for agent deployments.
+
+```yaml"
+grains:
+  web-client:
+    kind: kubernetes
+    spec:
+      sources:
+        ...
+      namespace: '{{ .inputs.namespace }}'
+```
+:::tip __Note__:
+* Launching concurrent sandboxes with Kubernetes grain is not supported for the same namespace. 
+* Make sure the Torque execution host has permissions to use those namespaces.
+:::
+
+## scripts (outputs)
+Due to Kubernetes deployment limitations, outputs need to be generated from a script and provided as part of the grain.
```

Comment:
Rephrase:
Kubernetes does not natively support outputs. Using Torque, you can overcome this and provide outputs from the manifest installation process using scripts.